### DISCUSSION TITLE
Keyboard shortcut to toggle transparent background

### DISCUSSION THREAD

**Discussion by MAXIIIMVS** [catppuccin/nvim#404] [Q&A] [answered]
Hi.
Thank you for building this beautiful color scheme.
I need a way to toggle between opaque and transparent backgrounds. But I couldn't find a way in the document.
Is there a way to change `transparent_background` programmatically?
Thanks

**Comment by EdenEast** [comment, marked as answer]
Here is an example solution:

```lua
vim.keymap.set("n", "<leader>tt", function()
  local cat = require("catppuccin")
  cat.options.transparent_background = not cat.options.transparent_background
  cat.compile()
  vim.cmd.colorscheme(vim.g.colors_name)
end)
```

Note that this will invalidate the cache file vs the compiled files. https://github.com/catppuccin/nvim/blob/a5f3ed5d3b1d9ea21183718a8a89a6653bd6ea48/lua/catppuccin/init.lua#L131-L154.

**Reply by MAXIIIMVS** [reply]
Thank you so much, this works.
